src/server/pfs/server/driver_file.go

```diff
@@ -118,18 +105,53 @@ func (d *driver) withWriter(pachClient *client.APIClient, commit *pfs.Commit, ta
 	if commitInfo.Finished != nil {
 		return pfsserver.ErrCommitFinished{commitInfo.Commit}
 	}
-	commit = commitInfo.Commit
-	n := d.getSubFileset()
-	subFileSetStr := fileset.SubFileSetStr(n)
-	if tag == "" {
-		tag = subFileSetStr
+	fsw := d.storage.NewWriter(ctx)
+	if err := cb(d.getDefaultTag(), fsw); err != nil {
+		if _, err := fsw.Close(); err != nil {
```

Comment:
I don't think we want to persist the changes if an error occurs. I think this is the case where we need the `Abort` function to just handle the clean up. For now, I think we would be better off making a note of this and not performing a `Close` when an error occurs.